code/__HELPERS/mobs.dm

```diff
@@ -81,4 +81,106 @@ proc/age2agedescription(age)
 		if(45 to 60)		return "middle-aged"
 		if(60 to 70)		return "aging"
 		if(70 to INFINITY)	return "elderly"
-		else				return "unknown"
\ No newline at end of file
+		else				return "unknown"
+
+/proc/do_mob(mob/user , mob/target, delay = 30, icon_display, prog_bar = PROGRESS_GENERIC, uninterruptible = FALSE, datum/callback/extra_checks)
+	if(!user || !target)
+		return FALSE
+	var/user_loc = user.loc
+
+	var/target_loc = target.loc
+
+	var/holding = user.get_active_held_item()
+	var/datum/progressbar/P = prog_bar ? new prog_bar(user, delay, target, icon_display) : null
+
+	user.action_busy++
+	var/endtime = world.time + delay
+	var/starttime = world.time
+	. = TRUE
+	while (world.time < endtime)
+		stoplag(1)
+		P?.update(world.time - starttime)
+
+		if(QDELETED(user) || QDELETED(target) || (extra_checks && !extra_checks.Invoke()))
+			. = FALSE
+			break
+		if(uninterruptible)
+			continue
+
+		if(user.loc != user_loc || target.loc != target_loc || user.get_active_held_item() != holding || user.incapacitated())
+			. = FALSE
+			break
+	if(P)
+		qdel(P)
+	user.action_busy--
+
+
+//some additional checks as a callback for for do_afters that want to break on losing health or on the mob taking action
+/mob/proc/break_do_after_checks(list/checked_health, check_clicks, selected_zone_check)
+	if(check_clicks && next_move > world.time)
+		return FALSE
+	if(selected_zone_check && zone_selected != selected_zone_check)
+		return FALSE
+	return TRUE
+
+//pass a list in the format list("health" = mob's health var) to check health during this
+/mob/living/break_do_after_checks(list/checked_health, check_clicks, selected_zone_check)
+	if(islist(checked_health))
+		if(health < checked_health["health"])
+			return FALSE
+		checked_health["health"] = health
+	return ..()
+
+/proc/do_after(mob/user, delay, needhand = TRUE, atom/target, icon_display, prog_bar = PROGRESS_GENERIC, datum/callback/extra_checks)
+	if(!user)
+		return FALSE
+
+	var/atom/Tloc
+	if(target && !isturf(target))
+		Tloc = target.loc
+
+	var/atom/Uloc = user.loc
+
+	var/holding = user.get_active_held_item()
+
+	var/holdingnull = TRUE //User's hand started out empty, check for an empty hand
```

Comment:
```suggestion
	var/holdingnull = holding ? FALSE : TRUE 
```

The comments for these lines don't really make sense, are they correct? If so can you clarify?

Reply:
Same feeling, this was pretty much copied from tg and is pretty much useless apparently, so removed this line and its usage however.